fix: HttpTransport does not catch built-in ConnectionError, leaking exceptions to callers

## Bug

`HttpTransport._send_with_retry` only catches `httpx.TimeoutException`, `httpx.NetworkError`, `TransientError`, `PermanentError`, and `ValueError`.

Python's built-in `ConnectionError` (an `OSError`) is **not** in this list. When a low-level connection failure raises the built-in `ConnectionError`, it bypasses the retry/swallow logic and propagates to the caller — breaking agent execution.

## Failing tests

```
FAILED tests/test_sdk_transport.py::test_transport_graceful_on_failure
FAILED tests/test_sdk_transport.py::test_transport_send_session_start_graceful_on_failure
FAILED tests/test_sdk_transport.py::test_transport_send_session_update_graceful_on_failure
```

## Root cause

`agent_debugger_sdk/transport.py:257` — the `except` tuple is too narrow.

## Fix

Broaden the `except` clause to `Exception` so any unexpected error is classified (via the existing `_classify_error` fallback) as permanent and swallowed with a log warning, rather than propagating.

--- a/agent_debugger_sdk/transport.py
+++ b/agent_debugger_sdk/transport.py
@@ -254,7 +254,7 @@ class HttpTransport:
             try:
                 await self._execute_request(method=method, path=path, payload=payload)
                 return
-            except (httpx.TimeoutException, httpx.NetworkError, TransientError, PermanentError, ValueError) as exc:
+            except Exception as exc:
                 last_error, should_retry = self._classify_error(exc)
 
                 if not should_retry:

